publiq-ui/pub-spinner.vue

```diff
@@ -0,0 +1,38 @@
+<template>
+  <div class="text-center d-flex justify-content-center spinner-container">
+    <b-spinner :small="isInButtonVariant" :variant="mappedVariant"></b-spinner>
+  </div>
+</template>
+
+<script>
+  const SpinnerVariant = {
+    onPage: 'primary',
+    inButton: 'light',
```

Comment:
Suggestion: Instead of the context (`onPage`, `inButton`) we could make it a prop that says more about the look of the spinner itself? Like `<pub-spinner color="primary">` and `<pub-spinner color="light">`.
Because I imagine we'll have other places we can use spinners than on a page or in a button.
Of course, it would be even better if those colors would be actual design tokens that are the same everywhere we use them.

Reply:
Also in a `secondary` button, you'd need a blue spinner. So then you end up with something like this with the `onPage` and `inButton` variants:

```
<pub-spinner variant="onPage" />

<pub-button variant="primary">
  <pub-spinner variant="inButton" />
</pub-button>

<pub-button variant="secondary">
  <pub-spinner variant="onPage" />
</pub-button>
```

Compare this to:

```
<pub-spinner variant="primary" />

<pub-button variant="primary">
  <pub-spinner variant="light" />
</pub-button>

<pub-button variant="secondary">
  <pub-spinner variant="primary" />
</pub-button>
```

Reply:
Additionally, we could add a `loading=true|false` prop to `pub-button` that would automatically show a spinner of the right variant for the button variant? It could also disable it automatically.

So we would only have to do this:
```
Blue button with white spinner:
<pub-button variant="primary" :loading="true"></pub-button>

White button with blue spinner:
<pub-button variant="secondary" :loading="true"></pub-button>
```

Reply:
Not only the color is the difference also the size of the `inButton` is smaller. But what do you think about this approach:
`<pub-spinner size="small" variant="light" :loading="true" />`
`<pub-spinner size="normal" variant="primary" :loading="true" />` 

`size="normal"` and `variant="primary"` would be the defaults if nothing is passed.

Reply:
Yeah btw your approach also can work, but that can be used in combination with my suggestion

Reply:
Yup good idea to add a size prop as well then. 👍 
The `:loading="true"` was meant for `pub-button` btw.